src/k8s/pkg/k8sd/controllers/node_configuration_test.go

```diff
@@ -0,0 +1,130 @@
+package controllers
+
+import (
+	"context"
+	"net"
+	"os"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/canonical/k8s/pkg/k8sd/setup"
+	"github.com/canonical/k8s/pkg/snap/mock"
+	snaputil "github.com/canonical/k8s/pkg/snap/util"
+	"github.com/canonical/k8s/pkg/utils/k8s"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes/fake"
+	k8stesting "k8s.io/client-go/testing"
+)
+
+func TestConfigPropagation(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	g := NewWithT(t)
+
+	dir := t.TempDir()
+
+	s := &mock.Snap{
+		Mock: mock.Mock{
+			KubernetesPKIDir:    path.Join(dir, "pki"),
```

Comment:
We keep setting up these standard dirs and IDs. Maybe create a `mock.NewDefaultMock(snapDir string)` method?

Reply:
Thanks for the review! I've checked tests in general and what kind of mocks required by a unit test varied a lot so just went with the simple approach. I suggest creating a separate card to improve/standardize mocks for all of the tests in this regard.

Reply:
fair